### ISSUE TITLE
Replace startup workaround with persistent listener

### ISSUE THREAD

**Issue by stoically** [mozilla/contain-facebook#206]
The workaround introduced in #78 due to the report in #61 to catch loading Facebook tabs on Firefox startup could potentially be replaced by "persistent listeners" introduced in [~~Bug 1447551~~](https://bugzilla.mozilla.org/show_bug.cgi?id=1447551).

> For a listener to become persistent, it must be added during evaluation/initial execution of the background script. If it happens later (asynchronously or otherwise) then it is not persisted.
- https://github.com/mdn/sprints/issues/1015

This would require to move the [webRequest listeners](https://github.com/mozilla/contain-facebook/blob/86cbe3c5a4bdf2abf9ef82cb9b7dd45579fa7861/src/background.js#L393-L405) to the global context and make sure to wait inside the listener for the required [init work](https://github.com/mozilla/contain-facebook/blob/86cbe3c5a4bdf2abf9ef82cb9b7dd45579fa7861/src/background.js#L376-L390). A short example is [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1501159#c12).

**Comment by Rob--W** [comment]
Someone reported this bug at https://bugzilla.mozilla.org/show_bug.cgi?id=1748999.

It looks like it should be straightforward to resolve this bug, by moving the `setupWebRequestListeners` call to the start of `init` at https://github.com/mozilla/contain-facebook/blob/058054e9ddb082a23a0272c25b634d54f599a121/src/background.js#L659-L676 .

**Comment by Myron-S** [comment]
Yep. Looks like this has been reported quite some time ago. I'm on 97.0b2 (64-bit) and this issue is present. I caught this because Facebook is logged in within its container, but not anywhere else so I get presented with the standard Facebook login page instead of the page referred to from the desktop shortcut.

Workaround is for me to start Firefox first and then invoke the shortcut on the desktop.
